dvc/utils/fs.py

```diff
@@ -63,3 +64,18 @@ def contains_symlink_up_to(path, base_path):
     if os.path.dirname(path) == path:
         return False
     return contains_symlink_up_to(os.path.dirname(path), base_path)
+
+
+def get_parent_dirs_up_to(dir, root_dir):
+    dir = os.path.normpath(dir)
+    root_dir = os.path.normpath(root_dir)
```

Comment:
you would probably need to make sure that root_dir and dir are absolute paths, right? Btw, please add a unit test for it.